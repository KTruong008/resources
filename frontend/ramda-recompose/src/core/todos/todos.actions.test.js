import {
  createTodo,
  broadcastCreatedTodo,
  fetchAllTodos,
  broadcastFetchedTodos,
  updateTodo,
  broadcastUpdatedTodo,
  deleteTodo,
  broadcastDeletedTodo,
} from './todos.actions';

describe('createTodo action', () => {
  it('should create the proper action object', () => {
    const action = createTodo();
    expect(action).toMatchSnapshot();
  });
});

describe('broadcastCreatedTodo action', () => {
  it('should create the proper action object', () => {
    const action = broadcastCreatedTodo({ id: 1, name: 'doYourWork' });
    expect(action).toMatchSnapshot();
  });
});

describe('fetchAllTodos action', () => {
  it('should create the proper action object', () => {
    const action = fetchAllTodos();
    expect(action).toMatchSnapshot();
  });
});

describe('broadcastFetchedTodos action', () => {
  it('should create the proper action object', () => {
    const action = broadcastFetchedTodos([{ id: 1, name: 'doYourWork' }]);
    expect(action).toMatchSnapshot();
  });
});

describe('updateTodo action', () => {
  it('should create the proper action object', () => {
    const action = updateTodo(1, { id: 1, name: 'updatedName' });
    expect(action).toMatchSnapshot();
  });
});

describe('broadcastUpdatedTodo action', () => {
  it('should create the proper action object', () => {
    const action = broadcastUpdatedTodo(2, { id: 2, name: 'updatedTodo' });
    expect(action).toMatchSnapshot();
  });
});

describe('deleteTodo action', () => {
  it('should create the proper action object', () => {
    const action = deleteTodo(1);
    expect(action).toMatchSnapshot();
  });
});

describe('broadcastDeletedTodo action', () => {
  it('should create the proper action object', () => {
    const action = broadcastDeletedTodo(1);
    expect(action).toMatchSnapshot();
  });
});
