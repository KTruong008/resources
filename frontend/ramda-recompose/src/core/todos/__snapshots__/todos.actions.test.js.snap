// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`broadcastCreatedTodo action should create the proper action object 1`] = `
Object {
  "data": Object {
    "id": 1,
    "name": "doYourWork",
  },
  "type": "CREATE_SUCCEEDED",
}
`;

exports[`broadcastDeletedTodo action should create the proper action object 1`] = `
Object {
  "todoId": 1,
  "type": "DELETE_SUCCEEDED",
}
`;

exports[`broadcastFetchedTodos action should create the proper action object 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "name": "doYourWork",
    },
  ],
  "type": "FETCH_SUCCEEDED",
}
`;

exports[`broadcastUpdatedTodo action should create the proper action object 1`] = `
Object {
  "data": Object {
    "id": 2,
    "name": "updatedTodo",
  },
  "todoId": 2,
  "type": "UPDATE_SUCCEEDED",
}
`;

exports[`createTodo action should create the proper action object 1`] = `
Object {
  "type": "CREATE",
}
`;

exports[`deleteTodo action should create the proper action object 1`] = `
Object {
  "todoId": 1,
  "type": "DELETE",
}
`;

exports[`fetchAllTodos action should create the proper action object 1`] = `
Object {
  "type": "FETCH",
}
`;

exports[`updateTodo action should create the proper action object 1`] = `
Object {
  "data": Object {
    "id": 1,
    "name": "updatedName",
  },
  "todoId": 1,
  "type": "UPDATE",
}
`;
